### PR TITLE
#165305736-Report an Article

### PR DESCRIPTION
**Pull request by Amoswachira**
#### What does this PR do?
* Reports an article that violates rules in the Authors heaven community
#### Description of Task to be completed?
* Ensure an authorized user can report an article
*Ensure an Admin user an view all reported articles
*Ensure that An Admin user can delete the reported articles
#### How should this be manually tested?
* After cloning the repo:
```
$ git checkout ft-report-article-165305736
$npm install
$ npm run start:dev 

*Report an article with a normal user.

*Login as an  Admin to delete any reported article. here are the credentials for Admin user:

    "email": "flex@gmail.com",
    "password": "QS@sjames7t6762r65r6y83@"


```
#### What are the relevant pivotal tracker stories?
[#165305736](https://www.pivotaltracker.com/n/projects/2326460/stories/165305736)
#### Screenshots
Screenshot 2019-06-25 at 12 20 01
Screenshot 2019-06-25 at 12 20 29
Screenshot 2019-06-25 at 12 16 56

### CODE REVIEW

File: src/redux/actions/postActions.js [hunk 1]
```diff
@@ -103,6 +108,55 @@ export const editPost = (slug, postData) => dispatch => {
...
+  }
+};
+
+export const getReports = () => dispatch => {
+  dispatch(setPostLoading());
+  return getreportsApi()
+    .then(res =>
+      dispatch({
+        type: GET_REPORTS,
+        payload: res.data
```
Comment (IssaIan) [inline src/redux/actions/postActions.js:147]: Catch network errors here for all the actions. When there is no network, user should receive a message.

**Review by IssaIan — CHANGES_REQUESTED** [review]

File: src/redux/actions/postActions.js (hunk 1 above)
Comment (Amoswachira) [reply src/redux/actions/postActions.js:147]: @IssaIan let me see how I can do that

Comment (Amoswachira) [reply src/redux/actions/postActions.js:147]: @IssaIan fixed

**Review by IssaIan — APPROVED** [review]

**Review by Philipotieno — CHANGES_REQUESTED** [review]
Solve the conflicts

**Comment by Amoswachira** [comment]
@Philipotieno let me work on that

**Comment by Amoswachira** [comment]
@Philipotieno I have rebased

**Review by Philipotieno — APPROVED** [review]

**Review by matthenge — APPROVED** [review]

File: src/redux/actions/postActions.js [hunk 2]
```diff
@@ -8,10 +10,13 @@ import {
...
   GET_PAGES,
-  GET_PAGES_NEXT
+  GET_PAGES_NEXT,
+  REPORT_POST,
+  GET_REPORTS,
+  DELETE_REPORT
 
 } from '../constants/types';
-import { postArticleApi, getPostsApi, getSingleArticle, deleteArticleApi, editArticle, getPagesApi, getNextPageApi } from '../../services/api';
+import { postArticleApi, getPostsApi, getSingleArticle, deleteArticleApi, editArticle, getPagesApi, getNextPageApi,reportArticleApi,getreportsApi,deleteReportApi } from '../../services/api';
```
Comment (Ogutu-Brian) [inline src/redux/actions/postActions.js]: Hello @Amoswachira ,
You could put the imports in new lines instead of having all of them in a long line

Comment (Amoswachira) [reply src/redux/actions/postActions.js]: @Ogutu-Brian okay

Comment (Amoswachira) [reply src/redux/actions/postActions.js]: @Ogutu-Brian it is fixed

**Review by Ogutu-Brian — APPROVED** [review]

**Pull request merged**